34 filter generated recipes by tags

mtariq-phd opened the pull request.

Commit:
fix: fix: update listeners to correctly track substitution and food group criteria, ensuring form validation correctly recognizes all user selections.

Commit:
fix: enable selecting default options and add reset button for recipe customization

- Allow users to select placeholder null options before recipe selection by ensuring form inputs are enabled and accessible
- Add a reset button to clear all custom recipe options, resetting selects and radios to default states
- Added CSS styling to make the reset button smaller than the generate button for better UI distinction
- Updated JavaScript to handle reset button behavior and reset internal state in RecipeOptionsManager

Commit:
fix: correct dietaryTags handling, clean code, improve customization UI, center generate button

- Updated dietaryTags logic to handle single-value selection instead of array
- Removed commented-out and unused legacy code for clarity
- Improved appearance and layout of recipe customization options
- Centered the generate-recipe button within the form for better UX

juangalbea reviewed (changes requested):
<img width="626" height="296" alt="Screenshot 2025-10-24 at 20 54 07" src="https://github.com/user-attachments/assets/322e8f4a-9e4c-4f27-bfda-248a2c91a8ea" />

Commit:
fix(recipe-generation): handle dietaryTags as array in API request

Fix the issue where dietaryTags was passed as an array to the backend,
causing a runtime error when calling .trim() on a non-string. Now
convert dietaryTags and other criteria to comma-separated strings
before sending in the POST request body.
